### DISCUSSION TITLE
currents.config.js with ESM

### DISCUSSION THREAD

**Discussion by samixchoumi** [currents-dev/cypress-currents-reporter#143] [Q&A] [answered]
Hello, I've set my `currents.config.js` file like that : 

```
// currents.config.js
module.exports = {
  projectId: "yyy",
  recordKey: "xxx",
  // Sorry Cypress users - set the director service URL
  cloudServiceUrl: "http://localhost:1234/",
};
```

And it seem that all the parameter are not recognized, I must set them into the CLI to make it work.
Otherly I will have an error message like this one : 

```
  currents:config failure details: Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/username/IdeaProjects/project/currents.config.js from /Users/username/IdeaProjects/project/node_modules/cypress-cloud/bin/cli.js not supported.
currents.config.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead rename currents.config.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in /Users/username/IdeaProjects/project/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).
 +6ms
 ERROR  Cannot resolve projectId. Please use one of the following:
- provide it as a "projectId" property for "run" API method
- set CURRENTS_PROJECT_ID environment variable
- set "projectId" in "currents.config.js" file

```
For information, my project use typescript.

Any idea on how to make it work properly ?

**Comment by agoldis** [comment, marked as answer]
You have "type: module" in your `package.json` - that forces the `require` statement to treat the module as ESM, but it's not an ESM. 

https://github.com/currents-dev/cypress-cloud/issues/125

**Comment by samixchoumi** [comment]
As you can see on the issue you send, I show why not using « type: module »
is wrong for me.
Gave me error to use cucumber preprocessor.
Is there any way to make it works ? @agoldis

**Comment by samixchoumi** [comment]
After I use the beta `1.8.0-beta.5`, I've a problem about the record key, it always told me that it's invalid. @agoldis 
The command I launch is this one : `CURRENTS_PROJECT_ID=xxx CURRENTS_RECORD_KEY=yyy DEBUG=currents:*,cypress:* npx cypress-cloud run --record --key yyy --parallel --ci-build-id $CI_PIPELINE_ID --env allure=true,configFile=dev --browser chrome`

And as I say before, all the data set into `currents.config.js` seem to not be detected.

The error I got : 
 ```
currents:api network request: { method: 'post', url: '/runs', headers: { Accept: 'application/json, text/plain, */*', 'Content-Type': 'application/json', 'x-cypress-request-attempt': 0, 'x-cypress-version': '12.13.0', 'x-ccy-version': '1.8.0-beta.5' }, data: { ci: { params: {}, provider: null }, specs: [ 'cypress/integration/Spec/AScenario.feature' ], commit: { branch: 'feature/update_sorry_cypress_to_cypress-cloud', remoteOrigin: 'git@gitlab.com:project/qat/projectName.git', authorEmail: 'samixchoumi@mail.com', authorName: 'samixchoumi', message: 'Update cypress-cloud config\n', sha: '73ca7974b08a93953bd176668f49e7bb303c30da' }, group: undefined, platform: { osName: 'darwin', osVersion: '22.5.0', osCpus: [Array], osMemory: [Object], browserName: 'Chrome', browserVersion: '114.0.5735.106' }, parallel: true, ciBuildId: '--env', projectId: 'xxx', recordKey: 'yyy', specPattern: [ './cypress/integration/**/*.feature' ], tags: [], testingType: 'e2e', batchSize: 3, autoCancelAfterFailures: undefined } } +0ms

 WARNING  Record key is invalid. Please visit https://app.currents.dev to see the valid record keys.

 ERROR  Request failed with status code 422
```

I don't understand why it redirect me to https://app.currents.dev while I use sorry-cypress ?

**Reply by agoldis** [reply]
Please open a new issue and follow the instruction to submit enough debug information - e.g. the complete debug log.

**Reply by samixchoumi** [reply]
https://github.com/currents-dev/cypress-cloud/issues/142 Opened this one, should I close this question then ?

**Reply by agoldis** [reply]
Just leave it as-is for future generations
